Duplicate identifier 'Action' in `node_modules/.prisma/client/index.d.ts`

## Bug description

When there is `model Action` in `schema.prisma`, the following error occurs when running `tsc` error

## How to reproduce

`schema.prisma`

```
model Action {
  id   String  @id @default(cuid())
  name String?
}
```

`Actions.ts`
```typescript
import { schema } from 'nexus'

schema.objectType({
  name: 'Action',
  definition(t) {
    t.model.id()
    t.model.name()
  }
})
```

run `npx prisma generate && tsc` An error occurred

```
node_modules/.prisma/client/index.d.ts:205:13-error TS2300: Duplicate identifier'Action'.
205 export type Action =
node_modules/.prisma/client/index.d.ts:987:13-error TS2300: Duplicate identifier'Action'
987 export type Action = {
```

`node_modules/.prisma/client/index.d.ts`

```
export type Action =
  | 'findOne'
  | 'findMany'
  | 'create'
  | 'update'
  | 'updateMany'
  | 'upsert'
  | 'delete'
  | 'deleteMany'
  | 'executeRaw'
  | 'queryRaw'
  | 'aggregate'
// ...

/**
 * Model Action
 */

export type Action = {
  id: string
  name: string
}
```

## Expected behavior

`Action` should not conflict with the type in prisma client

## Prisma information

`schema.prisma`
```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["aggregateApi", "distinct", "middlewares"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Action {
  id   String  @id @default(cuid())
  name String?
}
```

## Environment & setup

<!-- In which environment does the problem occur -->

- OS: Mac OS Catalina
- Database: PostgreSQL
- Node.js version: 14.7.0
- Prisma version:
```
@prisma/cli          : 2.4.1
Current platform     : darwin
Query Engine         : query-engine 195d4bdc2d16132977f4ba7a8ca312f7906cb086 (at node_modules/@prisma/cli/query-engine-darwin)
Migration Engine     : migration-engine-cli 195d4bdc2d16132977f4ba7a8ca312f7906cb086 (at node_modules/@prisma/cli/migration-engine-darwin)
Introspection Engine : introspection-core 195d4bdc2d16132977f4ba7a8ca312f7906cb086 (at node_modules/@prisma/cli/introspection-engine-darwin)
Format Binary        : prisma-fmt 195d4bdc2d16132977f4ba7a8ca312f7906cb086 (at node_modules/@prisma/cli/prisma-fmt-darwin)
Preview Features     : aggregateApi, distinct, middlewares
```

--- a/src/packages/client/src/generation/TSClient.ts
+++ b/src/packages/client/src/generation/TSClient.ts
@@ -288,12 +288,12 @@ export class TSClient implements Generatable {
 **/
 
 ${
-  this.options.platforms
-    ? this.options.platforms
-        .map((p) => `path.join(__dirname, 'query-engine-${p}');`)
-        .join('\n')
-    : ''
-}
+      this.options.platforms
+        ? this.options.platforms
+          .map((p) => `path.join(__dirname, 'query-engine-${p}');`)
+          .join('\n')
+        : ''
+      }
 
 /**
  * Annotation for \`node-file-trace\`
@@ -342,14 +342,14 @@ export declare type Datasource = {
   url?: string
 }
 ${new PrismaClientClass(
-  this.dmmf,
-  this.options.datasources,
-  this.options.outputDir,
-  this.options.browser,
-  this.options.generator,
-  this.options.sqliteDatasourceOverrides,
-  this.options.schemaDir,
-).toTS()}
+      this.dmmf,
+      this.options.datasources,
+      this.options.outputDir,
+      this.options.browser,
+      this.options.generator,
+      this.options.sqliteDatasourceOverrides,
+      this.options.schemaDir,
+    ).toTS()}
 
 ${/*new Query(this.dmmf, 'query')*/ ''}
 
@@ -363,16 +363,16 @@ ${/*new Query(this.dmmf, 'query')*/ ''}
 ${this.dmmf.schema.enums.map((type) => new Enum(type).toTS()).join('\n\n')}
 
 ${Object.values(this.dmmf.modelMap)
-  .map((model) => new Model(model, this.dmmf, this.options.generator!).toTS())
-  .join('\n')}
+        .map((model) => new Model(model, this.dmmf, this.options.generator!).toTS())
+        .join('\n')}
 
 /**
  * Deep Input Types
  */
 
 ${this.dmmf.inputTypes
-  .map((inputType) => new InputType(inputType).toTS())
-  .join('\n')}
+        .map((inputType) => new InputType(inputType).toTS())
+        .join('\n')}
 
 /**
  * Batch Payload for updateMany & deleteMany
@@ -392,7 +392,7 @@ export {};
 }
 
 class Datasources implements Generatable {
-  constructor(protected readonly internalDatasources: InternalDatasource[]) {}
+  constructor(protected readonly internalDatasources: InternalDatasource[]) { }
   public toTS(): string {
     const sources = this.internalDatasources
     return `export type Datasources = {
@@ -410,7 +410,7 @@ class PrismaClientClass implements Generatable {
     protected readonly generator?: GeneratorConfig,
     protected readonly sqliteDatasourceOverrides?: DatasourceOverwrite[],
     protected readonly cwd?: string,
-  ) {}
+  ) { }
   private get jsDoc(): string {
     const { dmmf } = this
 
@@ -500,7 +500,7 @@ export type LogEvent = {
 /* End Types for Logging */
 
 
-export type Action =
+export type PrismaAction =
   | 'findOne'
   | 'findMany'
   | 'create'
@@ -518,7 +518,7 @@ export type Action =
  */
 export type MiddlewareParams = {
   model?: string
-  action: Action
+  action: PrismaAction
   args: any
   dataPath: string[]
   runInTransaction: boolean
@@ -641,8 +641,8 @@ ${indent(this.jsDoc, tab)}
    */
   queryRaw<T = any>(query: string | TemplateStringsArray, ...values: any[]): Promise<T>;
 ${
-  this.generator?.previewFeatures?.includes('transactionApi')
-    ? `
+      this.generator?.previewFeatures?.includes('transactionApi')
+        ? `
   /**
    * Execute queries in a transaction
    * @example
@@ -660,18 +660,18 @@ ${
    */
   transaction: PromiseConstructor['all']
 `
-    : ''
-}
+        : ''
+      }
 ${indent(
-  dmmf.mappings
-    .filter((m) => m.findMany)
-    .map((m) => {
-      const methodName = lowerCase(m.model)
-      return `\
+        dmmf.mappings
+          .filter((m) => m.findMany)
+          .map((m) => {
+            const methodName = lowerCase(m.model)
+            return `\
 /**
  * \`prisma.${methodName}\`: Exposes CRUD operations for the **${
-        m.model
-      }** model.
+              m.model
+              }** model.
   * Example usage:
   * \`\`\`ts
   * // Fetch zero or more ${capitalize(m.plural)}
@@ -679,16 +679,16 @@ ${indent(
   * \`\`\`
   */
 get ${methodName}(): ${m.model}Delegate;`
-    })
-    .join('\n\n'),
-  2,
-)}
+          })
+          .join('\n\n'),
+        2,
+      )}
 }`
   }
 }
 
 class PayloadType implements Generatable {
-  constructor(protected readonly type: OutputType) {}
+  constructor(protected readonly type: OutputType) { }
 
   public toTS(): string {
     const { type } = this
@@ -731,19 +731,19 @@ export type ${getPayloadName(name)}<
     return `{
       [P in TrueKeys<S['${projection}']>]:${selectPrefix}
 ${indent(
-  relations
-    .map(
-      (f) => `P extends '${f.name}'
+      relations
+        .map(
+          (f) => `P extends '${f.name}'
 ? ${this.wrapType(
-        f,
-        `${getPayloadName(
-          (f.outputType.type as DMMF.OutputType).name,
-        )}<S['${projection}'][P]>`,
-      )} :`,
-    )
-    .join('\n'),
-  6,
-)} never
+            f,
+            `${getPayloadName(
+              (f.outputType.type as DMMF.OutputType).name,
+            )}<S['${projection}'][P]>`,
+          )} :`,
+        )
+        .join('\n'),
+      6,
+    )} never
     }`
   }
   private wrapType(field: DMMF.SchemaField, str: string): string {
@@ -812,7 +812,7 @@ export class Model implements Generatable {
     if (!aggregateType) {
       throw new Error(
         `Could not get aggregate type "${getAggregateName(model.name)}" for "${
-          model.name
+        model.name
         }"`,
       )
     }
@@ -848,70 +848,70 @@ export class Model implements Generatable {
       .join('\n')}
 
 ${
-  aggregateTypes.length > 1
-    ? aggregateTypes
-        .slice(1)
-        .map((type) => {
-          const newType: DMMF.InputType = {
-            name: getAggregateInputType(type.name),
-            fields: type.fields.map((field) => ({
-              ...field,
-              name: field.name,
-              inputType: [
-                {
-                  isList: false,
-                  isNullable: false,
-                  isRequired: false,
-                  kind: 'scalar',
-                  type: 'true',
-                },
-              ],
-            })),
-          }
-          return new InputType(newType).toTS()
-        })
-        .join('\n')
-    : ''
-}
+      aggregateTypes.length > 1
+        ? aggregateTypes
+          .slice(1)
+          .map((type) => {
+            const newType: DMMF.InputType = {
+              name: getAggregateInputType(type.name),
+              fields: type.fields.map((field) => ({
+                ...field,
+                name: field.name,
+                inputType: [
+                  {
+                    isList: false,
+                    isNullable: false,
+                    isRequired: false,
+                    kind: 'scalar',
+                    type: 'true',
+                  },
+                ],
+              })),
+            }
+            return new InputType(newType).toTS()
+          })
+          .join('\n')
+        : ''
+      }
 
 export type ${getAggregateArgsName(model.name)} = {
 ${indent(
-  aggregateRootField.args
-    .map((arg) => new InputField(arg).toTS())
-    .concat(
-      aggregateType.fields.map((f) => {
-        if (f.name === 'count') {
-          return `${f.name}?: true`
-        }
-        return `${f.name}?: ${getAggregateInputType(
-          (f.outputType.type as DMMF.OutputType).name,
-        )}`
-      }),
-    )
-    .join('\n'),
-  tab,
-)}
+        aggregateRootField.args
+          .map((arg) => new InputField(arg).toTS())
+          .concat(
+            aggregateType.fields.map((f) => {
+              if (f.name === 'count') {
+                return `${f.name}?: true`
+              }
+              return `${f.name}?: ${getAggregateInputType(
+                (f.outputType.type as DMMF.OutputType).name,
+              )}`
+            }),
+          )
+          .join('\n'),
+        tab,
+      )}
 }
 
 export type ${getAggregateGetName(model.name)}<T extends ${getAggregateArgsName(
-      model.name,
-    )}> = {
+        model.name,
+      )}> = {
   [P in keyof T]: P extends 'count' ? number : ${
-    aggregateTypes.length > 1
-      ? `${getAggregateScalarGetName(model.name)}<T[P]>`
-      : 'never'
-  }
+      aggregateTypes.length > 1
+        ? `${getAggregateScalarGetName(model.name)}<T[P]>`
+        : 'never'
+      }
 }
 
 ${
-  aggregateTypes.length > 1
-    ? `export type ${getAggregateScalarGetName(model.name)}<T extends any> = {
+      aggregateTypes.length > 1
+        ? `export type ${getAggregateScalarGetName(model.name)}<T extends any> = {
   [P in keyof T]: P extends keyof ${getAvgAggregateName(
-    model.name,
-  )} ? ${getAvgAggregateName(model.name)}[P] : never
+          model.name,
+        )} ? ${getAvgAggregateName(model.name)}[P] : never
 }`
-    : ''
-}
+        : ''
+      }
     
     `
   }
@@ -927,16 +927,16 @@ ${
     const includeType = hasRelationField
       ? `\nexport type ${getIncludeName(model.name)} = {
 ${indent(
-  outputType.fields
-    .filter((f) => f.outputType.kind === 'object')
-    .map(
-      (f) =>
-        `${f.name}?: boolean` +
-        (f.outputType.kind === 'object' ? ` | ${getFieldArgName(f)}` : ''),
-    )
-    .join('\n'),
-  tab,
-)}
+        outputType.fields
+          .filter((f) => f.outputType.kind === 'object')
+          .map(
+            (f) =>
+              `${f.name}?: boolean` +
+              (f.outputType.kind === 'object' ? ` | ${getFieldArgName(f)}` : ''),
+          )
+          .join('\n'),
+        tab,
+      )}
 }\n`
       : ''
 
@@ -947,27 +947,27 @@ ${indent(
 
 export type ${model.name} = {
 ${indent(
-  model.fields
-    .filter((f) => f.kind !== 'object')
-    .map((field) => new OutputField(field).toTS())
-    .join('\n'),
-  tab,
-)}
+      model.fields
+        .filter((f) => f.kind !== 'object')
+        .map((field) => new OutputField(field).toTS())
+        .join('\n'),
+      tab,
+    )}
 }
 
 ${this.getAggregationTypes()}
 
 export type ${getSelectName(model.name)} = {
 ${indent(
-  outputType.fields
-    .map(
-      (f) =>
-        `${f.name}?: boolean` +
-        (f.outputType.kind === 'object' ? ` | ${getFieldArgName(f)}` : ''),
-    )
-    .join('\n'),
-  tab,
-)}
+      outputType.fields
+        .map(
+          (f) =>
+            `${f.name}?: boolean` +
+            (f.outputType.kind === 'object' ? ` | ${getFieldArgName(f)}` : ''),
+        )
+        .join('\n'),
+      tab,
+    )}
 }
 ${includeType}
 ${new PayloadType(this.outputType!).toTS()}
@@ -1037,8 +1037,8 @@ const { count } = await ${method}({
       const onlySelect = firstScalar
         ? `\n// Only select the \`${firstScalar.name}\`
 const ${lowerCase(mapping.model)}With${capitalize(
-            firstScalar.name,
-          )}Only = await ${method}({ select: { ${firstScalar.name}: true } })`
+          firstScalar.name,
+        )}Only = await ${method}({ select: { ${firstScalar.name}: true } })`
         : ''
 
       return `Find zero or more ${plural}.
@@ -1146,7 +1146,7 @@ export class ModelDelegate implements Generatable {
     protected readonly outputType: OutputType,
     protected readonly dmmf: DMMFClass,
     protected readonly generator?: GeneratorConfig,
-  ) {}
+  ) { }
   public toTS(): string {
     const { fields, name } = this.outputType
     const mapping = this.dmmf.mappingsMap[name]
@@ -1164,33 +1164,33 @@ export class ModelDelegate implements Generatable {
     return `\
 export interface ${name}Delegate {
 ${indent(
-  actions
-    .map(
-      ([actionName]: [any, any]): string =>
-        `${getMethodJSDoc(actionName, mapping, model)}
+      actions
+        .map(
+          ([actionName]: [any, any]): string =>
+            `${getMethodJSDoc(actionName, mapping, model)}
 ${actionName}<T extends ${getModelArgName(name, actionName)}>(
   args${
-    actionName === DMMF.ModelAction.findMany ? '?' : ''
-  }: Subset<T, ${getModelArgName(name, actionName)}>
+            actionName === DMMF.ModelAction.findMany ? '?' : ''
+            }: Subset<T, ${getModelArgName(name, actionName)}>
 ): ${getSelectReturnType({ name, actionName, projection: Projection.select })}`,
-    )
-    .join('\n'),
-  tab,
-)}
+        )
+        .join('\n'),
+      tab,
+    )}
   /**
    * Count
    */
   count(args?: Omit<${getModelArgName(
-    name,
-    DMMF.ModelAction.findMany,
-  )}, 'select' | 'include'>): Promise<number>
+      name,
+      DMMF.ModelAction.findMany,
+    )}, 'select' | 'include'>): Promise<number>
 
   /**
    * Aggregate
    */
   aggregate<T extends ${getAggregateArgsName(
-    name,
-  )}>(args: Subset<T, ${getAggregateArgsName(
+      name,
+    )}>(args: Subset<T, ${getAggregateArgsName(
       name,
     )}>): Promise<${getAggregateGetName(name)}<T>>
 }
@@ -1217,28 +1217,28 @@ export declare class Prisma__${name}Client<T> implements Promise<T> {
   constructor(_dmmf: DMMFClass, _fetcher: PrismaClientFetcher, _queryType: 'query' | 'mutation', _rootField: string, _clientMethod: string, _args: any, _dataPath: string[], _errorFormat: ErrorFormat, _measurePerformance?: boolean | undefined, _isList?: boolean);
   readonly [Symbol.toStringTag]: 'PrismaClientPromise';
 ${indent(
-  fields
-    .filter((f) => f.outputType.kind === 'object')
-    .map((f) => {
-      const fieldTypeName = (f.outputType.type as DMMF.OutputType).name
-      return `
+      fields
+        .filter((f) => f.outputType.kind === 'object')
+        .map((f) => {
+          const fieldTypeName = (f.outputType.type as DMMF.OutputType).name
+          return `
 ${f.name}<T extends ${getFieldArgName(
-        f,
-      )} = {}>(args?: Subset<T, ${getFieldArgName(f)}>): ${getSelectReturnType({
-        name: fieldTypeName,
-        actionName: f.outputType.isList
-          ? DMMF.ModelAction.findMany
-          : DMMF.ModelAction.findOne,
-        hideCondition: false,
-        isField: true,
-        renderPromise: true,
-        fieldName: f.name,
-        projection: Projection.select,
-      })};`
-    })
-    .join('\n'),
-  2,
-)}
+            f,
+          )} = {}>(args?: Subset<T, ${getFieldArgName(f)}>): ${getSelectReturnType({
+            name: fieldTypeName,
+            actionName: f.outputType.isList
+              ? DMMF.ModelAction.findMany
+              : DMMF.ModelAction.findOne,
+            hideCondition: false,
+            isField: true,
+            renderPromise: true,
+            fieldName: f.name,
+            projection: Projection.select,
+          })};`
+        })
+        .join('\n'),
+      2,
+    )}
 
   private get _document();
   /**
@@ -1269,7 +1269,7 @@ export class InputField implements Generatable {
   constructor(
     protected readonly field: DMMF.SchemaArg,
     protected readonly prefixFilter = false,
-  ) {}
+  ) { }
   public toTS(): string {
     const { field } = this
     let fieldType
@@ -1280,8 +1280,8 @@ export class InputField implements Generatable {
           typeof t.type === 'string'
             ? GraphQLScalarToJSTypeTable[t.type] || t.type
             : this.prefixFilter
-            ? `Base${t.type.name}`
-            : t.type.name
+              ? `Base${t.type.name}`
+              : t.type.name
         type = JSOutputTypeToInputType[type] ?? type
         if (type === 'null') {
           hasNull = true
@@ -1308,7 +1308,7 @@ export class InputField implements Generatable {
 }
 
 export class OutputField implements Generatable {
-  constructor(protected readonly field: BaseField) {}
+  constructor(protected readonly field: BaseField) { }
   public toTS(): string {
     const { field } = this
     // ENUMTODO
@@ -1326,13 +1326,13 @@ export class OutputField implements Generatable {
 }
 
 export class SchemaOutputField implements Generatable {
-  constructor(protected readonly field: DMMF.SchemaField) {}
+  constructor(protected readonly field: DMMF.SchemaField) { }
   public toTS(): string {
     const { field } = this
     let fieldType =
       typeof field.outputType.type === 'string'
         ? GraphQLScalarToJSTypeTable[field.outputType.type] ||
-          field.outputType.type
+        field.outputType.type
         : field.outputType.type.name
     if (Array.isArray(fieldType)) {
       fieldType = fieldType[0]
@@ -1356,13 +1356,13 @@ export class SchemaOutputType implements Generatable {
     return `
 export type ${type.name} = {
 ${indent(
-  type.fields
-    .map((field) =>
-      new SchemaOutputField({ ...field, ...field.outputType }).toTS(),
-    )
-    .join('\n'),
-  tab,
-)}
+      type.fields
+        .map((field) =>
+          new SchemaOutputField({ ...field, ...field.outputType }).toTS(),
+        )
+        .join('\n'),
+      tab,
+    )}
 }`
   }
 }
@@ -1379,11 +1379,11 @@ export class OutputType implements Generatable {
     return `
 export type ${type.name} = {
 ${indent(
-  type.fields
-    .map((field) => new OutputField({ ...field, ...field.outputType }).toTS())
-    .join('\n'),
-  tab,
-)}
+      type.fields
+        .map((field) => new OutputField({ ...field, ...field.outputType }).toTS())
+        .join('\n'),
+      tab,
+    )}
 }`
   }
 }
@@ -1393,7 +1393,7 @@ export class MinimalArgsType implements Generatable {
     protected readonly args: DMMF.SchemaArg[],
     protected readonly model: DMMF.Model,
     protected readonly action?: DMMF.ModelAction,
-  ) {}
+  ) { }
   public toTS(): string {
     const { action, args } = this
     const { name } = this.model
@@ -1452,7 +1452,7 @@ export class ArgsType implements Generatable {
     protected readonly args: DMMF.SchemaArg[],
     protected readonly model: DMMF.Model,
     protected readonly action?: DMMF.ModelAction,
-  ) {}
+  ) { }
   public toTS(): string {
     const { action, args } = this
     const { name } = this.model
@@ -1509,29 +1509,29 @@ export class ArgsType implements Generatable {
  */
 export type ${getModelArgName(name, action)} = {
 ${indent(
-  bothArgsOptional.map((arg) => new InputField(arg).toTS()).join('\n'),
-  tab,
-)}
+      bothArgsOptional.map((arg) => new InputField(arg).toTS()).join('\n'),
+      tab,
+    )}
 }
 `
   }
 }
 
 export class InputType implements Generatable {
-  constructor(protected readonly type: DMMF.InputType) {}
+  constructor(protected readonly type: DMMF.InputType) { }
   public toTS(): string {
     const { type } = this
     const fields = uniqueBy(type.fields, (f) => f.name)
     // TO DISCUSS: Should we rely on TypeScript's error messages?
     const body = `{
 ${indent(
-  fields
-    .map((arg) =>
-      new InputField(arg /*, type.atLeastOne && !type.atMostOne*/).toTS(),
-    )
-    .join('\n'),
-  tab,
-)}
+      fields
+        .map((arg) =>
+          new InputField(arg /*, type.atLeastOne && !type.atMostOne*/).toTS(),
+        )
+        .join('\n'),
+      tab,
+    )}
 }`
     return `
 export type ${type.name} = ${body}`
@@ -1539,7 +1539,7 @@ export type ${type.name} = ${body}`
 }
 
 export class Enum implements Generatable {
-  constructor(protected readonly type: DMMF.Enum) {}
+  constructor(protected readonly type: DMMF.Enum) { }
   public toJS(): string {
     const { type } = this
     return `exports.${type.name} = makeEnum({
@@ -1555,7 +1555,7 @@ ${indent(type.values.map((v) => `${v}: '${v}'`).join(',\n'), tab)}
 
 export declare type ${type.name} = (typeof ${type.name})[keyof typeof ${
       type.name
-    }]\n`
+      }]\n`
   }
 }
 

